mobile/src/state/CurrentSchedule.ts

```diff
@@ -1,58 +1,28 @@
-import {useFetchJsonDebouncer} from "@/state/state-utilities";
-import {storeDailyCachedSchedule, useDailyCachedSchedule} from "@/state/CachedSchedules";
-import {match} from "ts-pattern";
-
-
-const CURRENT_SCHEDULE = ref<VoxxrinDailySchedule|undefined>(undefined);
-
-export const useCurrentSchedule = () => CURRENT_SCHEDULE.value;
-
-export const watchCurrentSchedule = (callback: (currentSchedule: (VoxxrinDailySchedule | undefined)) => void,) => {
-    watch(CURRENT_SCHEDULE, callback, {immediate: true});
-}
-
-export function unsetCurrentSchedule() {
-    CURRENT_SCHEDULE.value = undefined;
-}
-
-export const fetchSchedule = async (conferenceDescriptor: VoxxrinConferenceDescriptor, dayId: DayId) => {
-    // Avoiding to fetch schedule if the one already loaded matches the one expected
-    if(
-        !CURRENT_SCHEDULE.value?.eventId.isSameThan(conferenceDescriptor.id)
-        || !CURRENT_SCHEDULE.value?.day.isSameThan(dayId)
-    ) {
-        const day = findVoxxrinDay(conferenceDescriptor, dayId)
-        const voxxrinDailySchedule = await match(useDailyCachedSchedule(conferenceDescriptor.id, dayId))
-            .with(undefined, async () => {
-                const firestoreDailySchedule: DailySchedule = await useFetchJsonDebouncer(
-                    'daily-schedule',
-                    `events/${conferenceDescriptor.id.value}/days/${day.id.value}`,
-                    `/data/events/${conferenceDescriptor.id.value}/days/${day.id.value}.json`
-                );
-                console.debug(`timeslots fetched:`, firestoreDailySchedule.timeSlots)
-                const voxxrinSchedule = createVoxxrinDailyScheduleFromFirestore(conferenceDescriptor, firestoreDailySchedule);
-
-                // Caching schedule
-                storeDailyCachedSchedule(conferenceDescriptor.id, dayId, voxxrinSchedule);
-
-                return voxxrinSchedule;
-            })
-            .otherwise(async cachedDailySchedule =>
-                Promise.resolve(cachedDailySchedule.schedule())
-            )
-
-        CURRENT_SCHEDULE.value = voxxrinDailySchedule;
-    }
-
-    return CURRENT_SCHEDULE.value;
-}
+import {VoxxrinConferenceDescriptor} from "@/models/VoxxrinConferenceDescriptor";
+import {DocumentReference, doc} from "firebase/firestore";
+import {db} from "@/state/firebase";
+import {useFirestore} from "@vueuse/firebase";
+
+export function useSchedule(
+            conferenceDescriptorRef: Ref<VoxxrinConferenceDescriptor | undefined>, 
+            dayIdRef: Ref<DayId | undefined>) {
+
+    const document = computed(() => {
+        const conf = unref(conferenceDescriptorRef),
+            dayId = unref(dayIdRef);
+        return conf && dayId && 
+            (doc(db, `events/${conf.id.value}/days/${dayId.value}`) as DocumentReference<DailySchedule>)
+    })
+
+    const firestoreDailySchedule = useFirestore(document, undefined)
+
+    return computed(() => {
```

Comment:
@xhanin instead of returning a reactive object here, I'd stick to [Vue's recommandation](https://vuejs.org/guide/reusability/composables.html#conventions-and-best-practices) to return a plain (non-reactive) object with reactive props instead
> The recommended convention is for composables to always return a plain, non-reactive object containing multiple refs. This allows it to be destructured in components while retaining reactivity:

Reply:
unfortunately I tried it and it didn't work, or at least I wasn't able to destructure it as mentioned in the doc, with the example of x and y. But indeed it would be better to make it work like recommended.

Reply:
well, I've tried it again and it works, I'll push a fix commit